portable/ThirdParty/GCC/Posix/port.c

```diff
@@ -322,13 +368,21 @@ void vPortYield( void )
 
 void vPortDisableInterrupts( void )
 {
-    pthread_sigmask( SIG_BLOCK, &xAllSignals, NULL );
+    if( prvIsFreeRTOSThread( pthread_self() ) == pdFALSE )
+    {
+        return;
+    }
+    pthread_sigmask(SIG_BLOCK, &xAllSignals, NULL);
 }
 /*-----------------------------------------------------------*/
 
 void vPortEnableInterrupts( void )
 {
-    pthread_sigmask( SIG_UNBLOCK, &xAllSignals, NULL );
+    if( prvIsFreeRTOSThread( pthread_self() ) == pdFALSE )
+    {
+        return;
```

Comment:
Change logic to check if condition is true, remove the early return and instead just set the sigmask if true.